src/shared/auto-update-cache.ts

```diff
@@ -0,0 +1,58 @@
+import { TimeString } from '@types';
+import ms, { StringValue } from 'ms';
+type Timestamp = number;
+type CacheConstructorParams<Value> = {
+  calculate: () => Promise<Value>;
+  config: AutoUpdateCacheConfig;
+};
+
+const DEFAULT_RETRY_TIME: TimeString = '5M';
+
+export type AutoUpdateCacheConfig = {
+  valid?: 'always' | { onlyFor: StringValue };
+  update: { every: StringValue; ifFailsTryAgainIn?: StringValue };
```

Comment:
It's just a rename, but I think it's clearer
```suggestion
  valid?: 'always' | { onlyFor: TimeString };
  update: { every: TimeString; ifFailsTryAgainIn?: TimeString };
```